app/models/competition.rb

```diff
@@ -2490,23 +2499,33 @@ def set_form_data(form_data, current_user)
         if %w[registration.closingDateTime registration.waitingListDeadlineDate registration.eventChangeDeadlineDate].include?(joined_key)
           existing_value = current_state_form.dig(*prefixes, key)
 
-          if existing_value.present?
-            existing_datetime = DateTime.parse(existing_value).utc
+          previously_had_value = existing_value.present?
+          will_have_value = value.present?
 
-            raise WcaExceptions::BadApiParameter.new(I18n.t('competitions.errors.editing_deadline_already_passed', timestamp: existing_datetime), json_property: joined_key) unless existing_datetime >= DateTime.now.utc
+          # Complain if the existing timestamp lies in the past
+          #   Note: Some timestamps are less strict than others (see https://github.com/thewca/worldcubeassociation.org/issues/11416)
+          #   so we only enforce this validation on a subset of timestamps.
+          if previously_had_value && %w[registration.closingDateTime registration.waitingListDeadlineDate].include?(joined_key)
+            existing_datetime = DateTime.parse(existing_value)
 
-            if value.present?
-              new_datetime = DateTime.parse(value).utc
-              new_after_existing = new_datetime >= existing_datetime
+            raise WcaExceptions::BadApiParameter.new(I18n.t('competitions.errors.editing_deadline_already_passed', timestamp: existing_datetime), json_property: joined_key) if existing_datetime.past?
+          end
 
-              raise WcaExceptions::BadApiParameter.new(I18n.t('competitions.errors.edited_deadline_not_after_original', new_timestamp: new_datetime, timestamp: existing_datetime), json_property: joined_key) unless new_after_existing
-            end
+          if will_have_value
+            new_datetime = DateTime.parse(value)
+
+            # Complain if the new timestamp lies in the past
+            raise WcaExceptions::BadApiParameter.new(I18n.t('competitions.errors.edited_deadline_not_in_future', new_timestamp: new_datetime), json_property: joined_key) if new_datetime.past?
           end
 
-          if value.present?
-            new_datetime = DateTime.parse(value).utc
+          if previously_had_value && will_have_value
+            new_datetime = DateTime.parse(value)
+            existing_datetime = DateTime.parse(existing_value)
+
+            new_before_existing = new_datetime < existing_datetime
 
-            raise WcaExceptions::BadApiParameter.new(I18n.t('competitions.errors.edited_deadline_not_in_future', new_timestamp: new_datetime), json_property: joined_key) unless new_datetime > DateTime.now.utc
+            # Complain if the new value lies before the old value (the user is trying to move something into the past)
```

Comment:
This comment implies that the validation is trying to prevent deadlines being moved to the past. However, the logic seems to trigger if the new deadline is before the old deadline - irrespective of whether it is before/after the current date.

What is the desired behavior here?

Reply:
Okay, I can see how my comment was misleading. Will change that :)